src/SOS/SOS.Extensions/MemoryServiceFromDebuggerServices.cs

```diff
@@ -59,6 +59,8 @@ public bool ReadMemory(ulong address, Span<byte> buffer, out int bytesRead)
             HResult hr = _debuggerServices.ReadVirtual(address, buffer, out bytesRead);
             if (hr != HResult.S_OK)
             {
+                CheckCancellation(hr);
```

Comment:
The only thing I'm hesitant about is that the ReadMemory API doesn't normally thrown exceptions. At the minimum, the header in IMemoryService.cs needs to be updated with an exception entry.

Reply:
Does WinDBG ever have a handle directly to this method? If it does, we probably can't do this since it will crash things.

Reply:
Do you mean: is the ReadMemory API ever called in a managed interop function (called by C++)?  Like ReadVirtual in the various data target implementations? The answer is yes so this PR will crash windbg if a CTRL-C/cancel happens while reading memory in on the data targets (i.e. https://github.com/dotnet/diagnostics/blob/ef74d449a2aa27e45417928e448417183afbc5e0/src/SOS/SOS.Hosting/DataTargetWrapper.cs#L155)

So the more I think about, this PR isn't the right way to do this or it needs some more work.

Reply:
I never saw it crash anything when I tested it manually, but it's possible I didn't catch it in the right spots.  I had assumed that exceptions are marshalled as failed HRESULT across the boundary, but maybe that's an old Desktop CLR assumption from COM interop.  I can do some deeper testing to find out, if it's interesting.

I could simply set the Console.CancellationToken to be cancelled when we encounter this.  I've seen places where we keep executing past a `Console.CancellationToken.ThrowIfCancellationRequested()` even though Control-C was already hit and we've failed _multiple_ calls to ReadMemory.

I'll do some more digging here.

Reply:
Ah I think this is fine for windows, these are just vectored exceptions which will flow past native code.  Though obviously if they aren't expecting exceptions that might lead to issues.

The problem here is that only works on Windows, we can't unwind past native frames like that on Linux or OS X.  While those platforms don't return that particular error code from the data reader, it's too weird to do something very different on another platform.

I'll close this for now, but it would be nice if that return error code was immediately converted into a cancellation of Console.CancellationToken.  I don't see a good way to do this with the current design.  I'll leave that up to you all if you want to implement it.